consensus/src/network_tests.rs

```diff
@@ -339,6 +365,48 @@ impl NetworkPlayground {
             .split_network(partition_first, partition_second)
     }
 
+    /// Check if the message from 'src_twin_id' to 'dst_twin_id' should be dropped in the given round
+    pub fn is_message_dropped_round(&self, src: &TwinId, dst: &TwinId, round: u64) -> bool {
+        self.drop_config_round
+            .read()
+            .unwrap()
+            .is_message_dropped(src, dst, round)
+    }
+
+    /// Drop messages from 'src_twin_id' to 'dst_twin_id' in the given round
+    pub fn drop_message_for_round(&mut self, src: &TwinId, dst: &TwinId, round: u64) -> bool {
+        self.drop_config_round
+            .write()
+            .unwrap()
+            .drop_message_for(src, dst, round)
+    }
+
+    /// Creates the given per round network partitions
+    pub fn split_network_round(
+        &mut self,
+        round_partitions: &HashMap<u64, Vec<Vec<TwinId>>>,
+    ) -> bool {
+        let mut ret = true;
+
+        for (round, partitions) in round_partitions.iter() {
+            for (idx_curr_partition, curr_partition) in partitions.iter().enumerate() {
```

Comment:
this loop is quite intense, I think it just tries to achieve 
```
for i in 0..partitions.len() {
  for j in i+1..partitions.len() {
      self.drop_message_for_round(partitions[i], partitions[j], *round);
      ..
  }
}
```

Reply:
Not quite..

Reply:
sorry I misunderstood the loop, so here's what I got, the first two loops can also be replaced by map
```
for i in 0..partitions.len() {
  for j in i+1..partitions.len() {
    partitions[i].iter().flat_map(|src| partitions[j].iter().map(move |dst| (src, dst))).for_each(|(src, dst)| {
      self.drop_message_for_round(src, dst, *round);
      self.drop_message_for_round(dst, src, *round);
    })
  }
}
```

Reply:
```
    partitions
        .iter()
        .enumerate()
        .flat_map(|(i, v1)| {
            partitions.iter().skip(i + 1).flat_map(move |v2| {
                v1.iter()
                    .flat_map(move |src| v2.iter().map(move |dst| (src, dst)))
            })
        })
        .for_each(|(src, dst)| println!("{} {}", src, dst));
```

Reply:
Nice, thanks! Done.